gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -587,9 +610,52 @@ MM_MemoryPoolAddressOrderedList::collectorAllocate(MM_EnvironmentBase *env, MM_A
 	return addr;
 }
 
+/* Check the free entry being used and align it to appropriate boundaries for TLH allocation as required by GC.
+ * SATB Sweep has strict requirements for TLH to not span multiple Sweep Chunks. */
+bool
+MM_MemoryPoolAddressOrderedList::alignTLHForGC(MM_EnvironmentBase *env,  MM_HeapLinkedFreeHeader *freeEntry, uintptr_t *consumedSize)
+{
+	MM_GCExtensionsBase *extensions = env->getExtensions();
+
+	void *sweepChunkTopBoundary = NULL;
+	void *sweepChunkBaseBoundary = NULL;
+	void *tlhTopProjection = NULL;
+
+	if (NULL != _tlhAlignmentBoundary) {
+		Assert_MM_true(extensions->isSATBBarrierActive());
+
+		/* Determine the sweep chunk boundaries that contain the tlh being inited, we expect this chunk to be assigned during reassignChunks.*/
+		void* tlhBase = (void *)freeEntry;
+
+		Assert_MM_true(tlhBase >= _tlhAlignmentBoundary);
+
+		uintptr_t remainder = ((uintptr_t)tlhBase - (uintptr_t)_tlhAlignmentBoundary) % extensions->parSweepChunkSize;
```

Comment:
this method should be free of word 'sweep'. extensions->parSweepChunkSize.
it should be stored to another field like _tlhAlignmentSize (could be set by callback, every time we enable alignment, although effectively the value does not ever change)

I'd rename _tlhAlignmentBoundary to _tlhAlignmentBase